cmd/ddns-updater/main.go

```diff
@@ -167,6 +167,14 @@ func _main(ctx context.Context, reader *reader.Reader, args []string, logger log
 	client := &http.Client{Timeout: config.Client.Timeout}
 	defer client.CloseIdleConnections()
 
+	for _, p := range providers {
+		err = provider.Init(p, ctx, client)
+		if err != nil {
+			shoutrrrClient.Notify(err.Error())
+			return err
+		}
+	}
+
```

Comment:
Let's not do this, just force the user to set a domain name in the config.json, to match the existing architecture. There are also issues where for example the addrtools API does not work, then we won't know what domain name is failing (just the key) which isn't great.